File: src/main/java/com/yogesh/meditrack/service/DoctorService.java
package com.yogesh.meditrack.service;

import com.yogesh.meditrack.entity.Doctor;
import org.springframework.data.domain.Page;

import java.util.List;

public interface DoctorService {

    Doctor saveDoctor(Doctor doctor);

    List<Doctor> getAllDoctors();

    Doctor getDoctorById(Long id);

    Doctor updateDoctor(Long id, Doctor doctor);

    void deleteDoctor(Long id);

    List<Doctor> searchDoctor(String name);

    Page<Doctor> getDoctorsPage(
            int page,
            int size);
    List<Doctor> sortDoctors();
}

File: frontend/src/service/doctorService.js
import api from "./api";

export const getAllDoctors = () => api.get("/admin/doctors");

export const getDoctorById = (id) => api.get(`/admin/doctors/${id}`);

export const saveDoctor = (doctor) => api.post("/admin/doctors", doctor);

export const updateDoctor = (id, doctor) =>
  api.put(`/admin/doctors/${id}`, doctor);

export const deleteDoctor = (id) => api.delete(`/admin/doctors/${id}`);

export const searchDoctor = (name) =>
  api.get(`/admin/doctors/search`, { params: { name } });